### DISCUSSION TITLE
Convert color and grayscale in PDF to 1-bit black-and-white

### DISCUSSION THREAD

**Discussion by DavidGriffith** [ImageMagick/ImageMagick#5302] [General]
I am having a good deal of trouble with PDFs coming in that use lots of color for stuff that can easily be plain 1-bit black-and-white.  These documents are rather painful to read no matter if printed on a color or monochrome printer.  Color, because garish colors.  Monochrome, because the resulting grays (usually involving the type) look like the result of a failing toner cartridge.

Please tell me how to process these PDFs such that:
  1. All backgrounds are removed.
  2. Any color or gray is converted to 100-percent black.

**Comment by snibgo** [comment]
What version of IM, on what platform?

PDF files can contain vector data (including plain text and graphics) and raster images.

IM is a raster immage processor. It cn convert each PDF page into a raster image, and save those raster images into whatever format you want.

For example:
```
magick in.pdf -threshold 10% -fill White +opaque Black out.tiff
```

**Comment by DavidGriffith** [comment]
````
$ convert --version
Version: ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib
````

**Comment by snibgo** [comment]
For v6, use "convert" instead of "magick".

**Reply by xavier-au** [reply]
Try this, work fine for me with my HP880C black cartrige only.
convert -density 150  -type Grayscale -threshold 70%  /home/denis/Documents/file.pdf  /home/denis/Documents/outfile.pdf
